daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -395,27 +425,18 @@ func (devices *DeviceSet) deviceFileWalkFunction(path string, finfo os.FileInfo)
 		hash = ""
 	}
 
-	dinfo := devices.loadMetadata(hash)
-	if dinfo == nil {
-		return fmt.Errorf("Error loading device metadata file %s", hash)
+	// Include deleted devices also as cleanup delete device logic
+	// will go through it and see if there are any deleted devices.
+	if _, err := devices.lookupDevice(hash); err != nil {
```

Comment:
initMetaData() takes the devices.Lock().

initMetaData --> loadDeviceFilesOnStart() ----> deviceFileWalkFunction ---> lookupDevice()

There are couple of cases where I have lock already held and don't need a locked version
of lookupDevice(), hence I need both the versions.